ast/TreeCopying.cc

```diff
@@ -7,9 +7,9 @@ namespace sorbet::ast {
 
 namespace {
 
-TreePtr deepCopy(const Expression *avoid, const TreePtr &tree, bool root = false);
+TreePtr deepCopy(const void *avoid, const TreePtr &tree, bool root = false);
```

Comment:
Could this be a `TreePtr&`?

Reply:
I'll investigate a bit, this is definitely one of the blockers to making `TreePtr::get` private.

Reply:
OK, I investigated this a bit and found that the thing preventing this is that we often call the `deepCopy` member function from the individual tree nodes, where we won't have a `TreePtr` available to pass through. I think this will also be a problem for making the `TreePtr::get` function private as well :(